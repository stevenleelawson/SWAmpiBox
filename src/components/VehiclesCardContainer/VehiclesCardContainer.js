import React from 'react';
import VehicleCard from '../VehicleCard/VehicleCard';
import './VehiclesCardContainer.css'

const VehiclesCardContainer = ({data}) => {
  console.log('vehicle', data.results)
  let displayVehicles;
  if(data.results){
    displayVehicles = data.results.map((vehicle, index) => {
      return <VehicleCard name={vehicle.name}
                          model={vehicle.model}
                          vehicleClass={vehicle.vehicle_class}
                          passengers={vehicle.passengers}
                          key={vehicle.name + index}
                        />
    })
  } else {
    return null
  }
  return (
    <div className='card-container'>{displayVehicles}</div>
  )
}

export default VehiclesCardContainer;
