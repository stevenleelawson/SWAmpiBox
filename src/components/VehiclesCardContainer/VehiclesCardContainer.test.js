import React from 'react';
import { shallow } from 'enzyme';
import VehiclesCardContainer from './VehiclesCardContainer';

describe('VehiclesCardContainer', () => {
  it('should match the snapshot', () => {
    const mockData = {
      results: [ name: 'speeder' ]
    }
    const wrapper = shallow(<VehiclesCardContainer data={mockData}/>)
    expect(wrapper).toMatchSnapshot();
  })
})
