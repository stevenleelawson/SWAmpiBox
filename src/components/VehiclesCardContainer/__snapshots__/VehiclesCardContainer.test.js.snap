// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VehiclesCardContainer should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <VehiclesCardContainer
    data={
        Object {
            "results": Array [
              "nodejs",
            ],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <VehicleCard
          model={undefined}
          name={undefined}
          passengers={undefined}
          toggleFavorites={undefined}
          vehicleClass={undefined}
/>,
      ],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "NaN",
        "nodeType": "function",
        "props": Object {
          "model": undefined,
          "name": undefined,
          "passengers": undefined,
          "toggleFavorites": undefined,
          "vehicleClass": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <VehicleCard
            model={undefined}
            name={undefined}
            passengers={undefined}
            toggleFavorites={undefined}
            vehicleClass={undefined}
/>,
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "NaN",
          "nodeType": "function",
          "props": Object {
            "model": undefined,
            "name": undefined,
            "passengers": undefined,
            "toggleFavorites": undefined,
            "vehicleClass": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
